service/s3/api_op_WriteGetObjectResponse.go

```diff
@@ -369,6 +375,85 @@ func (c *Client) addOperationWriteGetObjectResponseMiddlewares(stack *middleware
 	return nil
 }
 
+type opWriteGetObjectResponseResolveEndpointMiddleware struct {
+	EndpointResolver               EndpointResolverV2
+	BuiltInResolver                endpoints.BuiltInParameterResolver
+	ForcePathStyle                 *bool
+	Accelerate                     *bool
+	DisableMultiRegionAccessPoints *bool
+	UseArnRegion                   *bool
+}
+
+func (*opWriteGetObjectResponseResolveEndpointMiddleware) ID() string {
+	return "opWriteGetObjectResponseResolveEndpointMiddleware"
+}
+
+func (m *opWriteGetObjectResponseResolveEndpointMiddleware) HandleSerialize(ctx context.Context, in middleware.SerializeInput, next middleware.SerializeHandler) (
+	out middleware.SerializeOutput, metadata middleware.Metadata, err error,
+) {
+	req, ok := in.Request.(*smithyhttp.Request)
+	if !ok {
+		return out, metadata, fmt.Errorf("unknown transport type %T", in.Request)
+	}
+
+	if m.EndpointResolver == nil {
+		return out, metadata, fmt.Errorf("expected endpoint resolver to not be nil")
+	}
+
+	if m.BuiltInResolver == nil {
+		m.BuiltInResolver = &endpoints.NopBuiltInResolver{}
+	}
+
+	params := EndpointParameters{}
+
+	resolveBuiltIns(params, m.BuiltInResolver)
+
+	params.UseObjectLambdaEndpoint = aws.Bool(true)
+
+	var resolvedEndpoint smithyendpoints.Endpoint
+	resolvedEndpoint, err = m.EndpointResolver.ResolveEndpoint(ctx, params)
+	if err != nil {
+		return out, metadata, fmt.Errorf("failed to resolve service endpoint, %w", err)
+	}
+
+	req.URL = &resolvedEndpoint.URI
+
+	auth, ok := resolvedEndpoint.Properties.Get("authSchemes").([]interface{})
+	if ok {
+		for _, schemes := range auth {
+			scheme, ok := schemes.(map[string]interface{})
+			if !ok {
+				continue
+			}
+			if len(awsmiddleware.GetSigningName(ctx)) == 0 {
```

Comment:
**fix**: We need to account for all the properties in `authScheme` including `signingRegion/signingRegions` (v4 is singular, v4a is a list) as well as the `signingService` name.